src/App.jsx

```diff
@@ -13,90 +15,83 @@ import {
   SinglePost,
   UserProfile,
 } from "./pages";
+import { Footer } from "./components";
 
 function App() {
   return (
-    <Routes>
-      <Route
-        path="login"
-        element={
-          <RedirectLoggedInUser>
-            <Login />
-          </RedirectLoggedInUser>
-        }
-      />
-      <Route
-        path="signup"
-        element={
-          <RedirectLoggedInUser>
-            <Signup />
-          </RedirectLoggedInUser>
-        }
-      />
-      <Route path="/" element={<Navigate to={"feed"} />} />
-      <Route
-        path="feed"
-        element={
-          <ProtectedRoute>
+    <div className="App">
+      <ToastContainer />
+      <Routes>
+        <Route
+          path="login"
+          element={
+            <RedirectLoggedInUser>
+              <Login />
+            </RedirectLoggedInUser>
+          }
+        />
+        <Route
+          path="signup"
+          element={
+            <RedirectLoggedInUser>
+              <Signup />
+            </RedirectLoggedInUser>
+          }
+        />
```

Comment:
You can enclose all the protected routes within a single Component, instead of enclosing it at every path.